src/main/java/revai/StreamingClient.java

```diff
@@ -0,0 +1,205 @@
+
+  /**
+   * Sends an HTTP request and upon authorization is upgraded to a websocket connection. Use
+   * websocket listener to handle websocket events.
+   *
+   * @param revAiWebsocketListener
+   * @see WebSocketListener
+   * @throws URISyntaxException
+   */
+  public void connect(RevAiWebsocketListener revAiWebsocketListener) throws URISyntaxException {
+    Listener listener = new Listener(revAiWebsocketListener);
+    Request request = new Request.Builder().url(buildURL()).build();
+    webSocket = client.newWebSocket(request, listener);
+  }
+
+  /**
+   * Sends data over websocket in the form of a ByteString
+   *
+   * @param byteString
+   */
+  public void sendAudioData(ByteString byteString) {
+    webSocket.send(byteString);
+  }
+
+  /** Will close the websocket connection by informing the server that it's the End of Stream */
+  public void close() {
+    webSocket.send("EOS");
+  }
+
+  private String buildURL() throws URISyntaxException {
+    URIBuilder uriBuilder = new URIBuilder();
+    if (scheme != null) {
+      uriBuilder.setScheme(scheme);
+    } else {
+      uriBuilder.setScheme("wss");
+    }
+    if (host != null) {
+      uriBuilder.setHost(host);
+    } else {
+      uriBuilder.setHost("api.rev.ai");
+    }
+    uriBuilder.setPath("/speechtotext/v1/stream");
+    uriBuilder.setParameter("access_token", accessToken);
+    if (metadata != null) {
+      uriBuilder.setParameter("metadata", metadata);
+    }
+    if (customVocabularyId != null) {
+      uriBuilder.setParameter("custom_vocabulary_id", customVocabularyId);
+    }
+    if (filterProfanity != null) {
+      uriBuilder.setParameter("filter_profanity", String.valueOf(filterProfanity));
+    }
+    return uriBuilder.build().toString() + buildContentString();
+  }
+
+  private String buildContentString() {
+    List<String> content = getListFromContentType();
+    String empty = "";
+    if (content.size() == 0) {
+      return empty;
+    } else {
+      return createContentTypeParameter(content);
+    }
+  }
+
+  private List<String> getListFromContentType() {
+    List<String> content = new ArrayList<>();
+    if (streamContentType.getContentType() != null) {
+      content.add("content_type" + "=" + streamContentType.getContentType());
+    }
+    if (streamContentType.getLayout() != null) {
+      content.add("layout" + "=" + streamContentType.getLayout());
+    }
+    if (streamContentType.getRate() != null) {
+      content.add("rate" + "=" + streamContentType.getRate());
+    }
+    if (streamContentType.getFormat() != null) {
+      content.add("format" + "=" + streamContentType.getFormat());
+    }
+    if (streamContentType.getChannels() != null) {
+      content.add("channels" + "=" + streamContentType.getChannels());
+    }
+    return content;
+  }
+
+  private String createContentTypeParameter(List<String> content) {
+    String completeContentType = content.stream().collect(Collectors.joining(";"));
+    return "&" + completeContentType;
```

Comment:
super minor: i think you can just do `String.join(";", content)`, slightly simpler.
also minor / pedantic, but i would return it without the `&contentType=` - instead in buildUrl I would be 
`return uriBuilder.build().toString() + "&content_type=buildContentString()`.

The reason is - the value of the content type is "[type];layout=foo;rate=x" etc. then you are passing that as the value of the `content_type` query parameter when building the url. so it makes sense to keep that separation. In fact i would move the construction of the content type string to a method on `StreamContentType`, similar as in the node SDK (https://github.com/revdotcom/revai-node-sdk/blob/401af632ec3e89c6051b3ff9812963c0b484d5fe/src/models/streaming/AudioConfig.ts#L32)